FuelLabs/fuel-indexer
# Fix from opts env var injection

### Description
- Fixes an issue where env vars were not properly being injected in the config
  - The opt env vars were being read properly (e.g., `--postgres-password $POSTGRES_PASSWORD`) but general env vars (e.g., `POSTGRES_PASSWORD=password cargo run`) were not
  - Apparently this general env var setup is needed for https://github.com/FuelLabs/fuel-indexer/pull/212
- Limits env vars to just postgres options
- Renames some things for clarity

### Testing steps
- [ ] Try running the service with env vars

```bash
POSTGRES_PASSWORD=postgres POSTGRES_HOST=0.0.0.0 RUST_LOG=INFO cargo run \
    --bin fuel-indexer -- --manifest examples/block-explorer/manifest.yaml
```

Should output

```bash
➜  fuel-indexer git:(rashad/fix-env-var-injection-from-opts) ✗ POSTGRES_HOST=0.0.0.0 POSTGRES_PASSWORD=password RUST_LOG=INFO cargo run --bin fuel-indexer -- --manifest examples/block-explorer/manifest.yaml
    Blocking waiting for file lock on build directory
   Compiling fuel-indexer-lib v0.1.0 (/Users/rashad/Development/repos/fuel-indexer/fuel-indexer-lib)
   Compiling fuel-indexer-schema v0.0.0 (/Users/rashad/Development/repos/fuel-indexer/fuel-indexer-schema)
   Compiling fuel-indexer v0.0.0 (/Users/rashad/Development/repos/fuel-indexer/fuel-indexer)
    Finished dev [unoptimized + debuginfo] target(s) in 16.27s
     Running `target/debug/fuel-indexer --manifest examples/block-explorer/manifest.yaml`
Sep 14 08:55:41.979  INFO fuel_indexer: Configuration: IndexerConfig { fuel_node: FuelNodeConfig { host: "127.0.0.1", port: "4000" }, graphql_api: GraphQLConfig { host: "0.0.0.0", port: "29987", run_migrations: None }, database: PostgresConfig { user: "postgres", password: "XXXX", host: "0.0.0.0", port: "5432", database: "postgres" } }
```

- [ ] Try running with opt env vars

```bash
cargo run --bin fuel-indexer -- --postgres-password $POSTGRES_PASSWORD
```

Should output

```bash
➜  fuel-indexer git:(rashad/fix-env-var-injection-from-opts) ✗ RUST_LOG=INFO cargo run --bin fuel-indexer -- --manifest examples/block-explorer/manifest.yaml --postgres-host $POSTGRES_HOST --postgres-password $POSTGRES_PASSWORD
    Finished dev [unoptimized + debuginfo] target(s) in 0.62s
     Running `target/debug/fuel-indexer --manifest examples/block-explorer/manifest.yaml --postgres-host 0.0.0.0 --postgres-password postgres`
Sep 14 08:56:09.554  INFO fuel_indexer: Configuration: IndexerConfig { fuel_node: FuelNodeConfig { host: "127.0.0.1", port: "4000" }, graphql_api: GraphQLConfig { host: "0.0.0.0", port: "29987", run_migrations: None }, database: PostgresConfig { user: "postgres", password: "XXXX", host: "0.0.0.0", port: "5432", database: "postgres" } }
```
